### PR TITLE
[5.8] Throw Error If Transaction Is Not Started While Commit

### PR DESCRIPTION
**Pull request by aman00323**
This pull request has the solution to bug **User commit when there is no transaction started**
The current code never notify user that transaction is not started and they are trying to commit.

**Current 5.8 Code**
```php
        if ($this->transactions == 1) {
            $this->getPdo()->commit();
        }
        $this->transactions = max(0, $this->transactions - 1);
        $this->fireConnectionEvent('committed');
```

This could be disaster if someone forget to start transaction and they trying to commit, So changes in code will throw exception. That will indicate developer that they are trying to commit but no transaction has started yet.

**Changes done to code**
```php
        if ($this->transactions == 1) {
            $this->getPdo()->commit();
        } elseif ($this->transactions == 0) {
            throw new TransactionException;
        } else {
            $this->transactions = max(0, $this->transactions - 1);
            $this->fireConnectionEvent('committed');
        }
```

This issue is already reported here https://github.com/laravel/framework/issues/29505, and also this issue was mention earlier by @GrahamCampbell 

@taylorotwell  have look at this issue and it's PR.

### CODE REVIEW

File: src/Illuminate/Database/Concerns/ManagesTransactions.php [hunk 1]
```diff
@@ -155,11 +156,12 @@ public function commit()
     {
         if ($this->transactions == 1) {
             $this->getPdo()->commit();
+        } elseif ($this->transactions == 0) {
+            throw new TransactionException;
```
Comment (mfn) [inline src/Illuminate/Database/Concerns/ManagesTransactions.php]: I suggest to add a message here, given more detail why it can't be commited (because none was started)

**Review by mfn — COMMENTED** [review]
I'm still not convinced this should be pushed into a minor release (ref: https://github.com/laravel/framework/pull/29649#issuecomment-523334366 ) due to the breaking change.

Even if it's currently wrong, it's never nice to break peoples app during a minor upgrade and it seems this behaviour was there for a long time so there is no "rush" (at 6.0 is around the corner was we all know).

TL;DR: IMHO this is better suited for 6.0

**Comment by deleugpn** [comment]
High potential for breaking several applications on a point release.
